migrate/indexerdb0to1.js

```diff
@@ -0,0 +1,221 @@
+
+  await checkVersion(db, version - 1);
+
+  const buf = Buffer.allocUnsafe(5 + 4);
+  buf.write('chain', 0, 'ascii');
+  buf.writeUInt32LE(version, 5, true);
+
+  const parent = db.batch();
+  parent.put(layout.V.encode(), buf);
+  await parent.write();
+}
+
+async function checkVersion(db, version) {
+  console.log('Checking version.');
+
+  const ver = await getVersion(db);
+
+  if (ver !== version)
+    throw Error(`DB is version ${ver}.`);
+
+  return ver;
+}
+
+async function migrateFilter(db) {
+  console.log('Migrating filters..');
+  let parent  = db.batch();
+
+  const iter = db.iterator({
+    gte: layout.h.min(),
+    lte: layout.h.max(),
+    keys: true,
+    values: true
+  });
+
+  const migratedDB = bdb.create({
+    location: path.join(db.location, 'BASIC'),
+    memory: false,
+    compression: true,
+    createIfMissing: true
+  });
+
+  await migratedDB.open();
+
+  let parentMigratedDB = migratedDB.batch();
+
+  let migratedFilters = 0;
+
+  await iter.each(async (height, hash) => {
+    const rawFilter = await blockStore.readFilter(hash, filters.BASIC);
+    const filter = Filter.fromRaw(rawFilter);
+    const filterHash = hash256.digest(filter.filter);
+
+    parentMigratedDB.put(height, hash);
+    parent.del(height);
+
+    parentMigratedDB.put(layout.f.encode(hash), filterHash);
+    parent.del(layout.f.encode(hash));
+
+    migratedFilters += 1;
+
+    if (migratedFilters % 10000 === 0) {
+      console.log('migrated %d filters.', migratedFilters);
+      await parentMigratedDB.write();
+      await parent.write();
+
+      parentMigratedDB = migratedDB.batch();
+      parent = db.batch();
+    }
+  });
+
+  let raw = await db.get(layout.R.encode());
+  parentMigratedDB.put(layout.R.encode(), raw);
+  parent.del(layout.R.encode());
+
+  raw = await db.get(layout.O.encode());
+  parentMigratedDB.put(layout.O.encode(), raw);
+  parent.del(layout.O.encode());
+
+  await updateVersion(db, 1);
+  raw = await db.get(layout.V.encode());
+  parentMigratedDB.put(layout.V.encode(), raw);
+  parent.del(layout.V.encode());
+  console.log('Filter DB updated to version 1');
+
+  await parentMigratedDB.write();
+  await parent.write();
+
+  console.log('%d filters migrated.', migratedFilters);
+
+  await db.close();
+  await db.destroy();
+
+  await migratedDB.close();
+}
+
+/*
+ * Execute
+ */
+
+let count = 0;
+
+(async () => {
+  const indexesLocation = path.resolve(process.argv[2]);
+  const indexes = ['tx', 'addr', 'filter'];
+
+  // Updating chainDB version to ensure
+  // filter indexes are not duplicated.
+  const chainDB = bdb.create({
+    location: path.resolve(process.argv[2], '../chain'),
+    compression: true,
+    cacheSize: 32 << 20,
+    createIfMissing: false
+  });
+
+  console.log('Opened %s.', path.resolve(process.argv[2], '../chain'));
+  await chainDB.open();
+
+  const chainVersion = await getVersion(chainDB);
+  if (chainVersion <= 6) {
+    await updateChainVersion(chainDB, 7);
+  }
```

Comment:
Just in case someone's DB is really old, we want to make sure they do the 4to6 migration FIRST. But if they are already on version 7 then yeah, nothing to do here.

```suggestion
  if (chainVersion === 6) {
    await updateChainVersion(chainDB, 7);
  }
```